spec/requests/api/users_spec.rb

```diff
@@ -0,0 +1,113 @@
+describe 'Users API' do
+  # TODO share common functionality w/ other API specs
+
+  def get_json(url)
+    get(url)
+    JSON.parse(response.body)
+  end
+
+  def time_to_s(time)
+    time.iso8601(3)
+  end
+
+  before do
+    # TODO clean this up
+    ENV['API_ENABLED'] = 'true'
+  end
+
+  describe 'GET /api/v1/users.json' do
+    it "responds with the list of users" do
+      user = FactoryGirl.create(:user)
+      json = get_json('/api/v1/users.json')
+      expect(json).to eq([
+        {
+          'created_at' => time_to_s(user.created_at),
+          'id' => user.id,
+          'updated_at' => time_to_s(user.updated_at)
+        }
+      ])
+    end
+
+    it "includes the personal details when signed in" do
+      user = FactoryGirl.create(:user)
+      login_as(user)
+
+      json = get_json('/api/v1/users.json')
+      expect(json).to eq([
+        {
+          'created_at' => time_to_s(user.created_at),
+          'email_address' => user.email_address,
+          'first_name' => user.first_name,
+          'id' => user.id,
+          'last_name' => user.last_name,
+          'updated_at' => time_to_s(user.updated_at)
+        }
+      ])
+    end
+
+    it "responds with an empty list for no users" do
+      json = get_json('/api/v1/users.json')
+      expect(json).to eq([])
+    end
+
+    it "can be `limit`ed" do
+      3.times do
+        FactoryGirl.create(:user)
+      end
+
+      json = get_json('/api/v1/users.json?limit=2')
+      expect(json.size).to eq(2)
+    end
+
+    it "can be `offset`" do
+      users = 3.times.map do
+        FactoryGirl.create(:user)
+      end
+
+      json = get_json('/api/v1/users.json?offset=1')
+
+      ids = json.map {|user| user['id'] }
+      expect(ids).to eq(users.map(&:id)[1..-1])
```

Comment:
It doesn't look like a sort is involved here? Does offset make sense without a sort?